packages/console-client/src/client/http/axios/axiosRequestAdapter.ts

```diff
@@ -0,0 +1,174 @@
+    return response.data
+  }
+
+  public sendCollectionOfPrimitive<ResponseType extends Exclude<PrimitiveTypesForDeserializationType, ArrayBuffer>>(
+    requestInfo: RequestInformation,
+    responseType: Exclude<PrimitiveTypesForDeserialization, 'ArrayBuffer'>,
+    errorMappings: ErrorMappings | undefined
+  ): Promise<ResponseType[] | undefined> {
+    throw new Error('Method not implemented.')
+  }
+
+  public sendPrimitive<ResponseType extends PrimitiveTypesForDeserializationType>(
+    requestInfo: RequestInformation,
+    responseType: PrimitiveTypesForDeserialization,
+    errorMappings: ErrorMappings | undefined
+  ): Promise<ResponseType | undefined> {
+    throw new Error('Method not implemented.')
+  }
+
+  public sendNoResponseContent(
+    requestInfo: RequestInformation,
+    errorMappings: ErrorMappings | undefined
+  ): Promise<void> {
+    throw new Error('Method not implemented.')
+  }
+
+  public sendEnum<EnumObject extends Record<string, unknown>>(
+    requestInfo: RequestInformation,
+    enumObject: EnumObject,
+    errorMappings: ErrorMappings | undefined
+  ): Promise<EnumObject[keyof EnumObject] | undefined> {
+    throw new Error('Method not implemented.')
+  }
+
+  public sendCollectionOfEnum<EnumObject extends Record<string, unknown>>(
+    requestInfo: RequestInformation,
+    enumObject: EnumObject,
+    errorMappings: ErrorMappings | undefined
+  ): Promise<EnumObject[keyof EnumObject][] | undefined> {
+    throw new Error('Method not implemented.')
+  }
+
+  public enableBackingStore(
+    backingStoreFactory?: BackingStoreFactory | undefined
+  ): void {
+    throw new Error('Method not implemented.')
+  }
+
+  public convertToNativeRequest<T>(
+    requestInfo: RequestInformation): Promise<T> {
+    throw new Error('Method not implemented.')
+  }
+
+  private async getHttpResponseMessage<T>(requestInfo: RequestInformation, claims?: string): Promise<AxiosClientResponse<T>> {
+    InvalidArgumentError.AssertNotFalsy('requestInfo', requestInfo)
+
+    this.setBaseUrlForRequestInformation(requestInfo)
+    const requestConfig = await this.getRequestConfigFromRequestInformation(requestInfo)
+    const response = await this.httpClient.executeAxios<T>(requestInfo.URL, requestConfig, requestInfo.getRequestOptions())
+
+    return response
+  }
+
+  private setBaseUrlForRequestInformation = (requestInfo: RequestInformation): void => {
+    InvalidArgumentError.AssertNotFalsy('baseUrl', this.baseUrl)
+
+    requestInfo.pathParameters.baseurl = this.baseUrl
+  }
+
+  private getRequestConfigFromRequestInformation = async(requestInfo: RequestInformation): Promise<AxiosClientRequestConfig> => {
+    this.setRequestInformationDefaultHeaders(requestInfo)
+
+    const method = requestInfo.httpMethod?.toString()
+    const headers: [string, string][] | undefined = requestInfo.headers
+      ? Array
+        .from(requestInfo.headers.keys())
+        .map((key) => [
```

Comment:
Is it good to return an array of the array here?
You can use second params to array from to avoid using ".map" after like this 

```
Array.from(requestInfo.headers.keys(), (key) => [key.toString().toLocaleLowerCase(), ...])
```

Reply:
in the case expected, why not directly use a reduce method?

Reply:
in this case `requestInfo.headers` is the kiota abstraction over the concept of headers. under the hood it is implemented as a class that extends Map type `declare class Headers extends Map<string, Set<string>>` for this reason it is not directly iterable with a .reduce  

Conceirning the first suggestion, what's the real difference in using the Array.from with mapFn param instead of .map method chaining?
(asking as to me it seems the same thing, just that with method chaining is a bit more clear and more explicit)